titus-common/src/main/java/com/netflix/titus/common/util/rx/RetryHandlerBuilder.java

```diff
@@ -118,6 +125,13 @@ public RetryHandlerBuilder but() {
                 .doOnNext(error -> onErrorHook.call(error))
                 .zipWith(Observable.range(0, retryCount + 1), RetryItem::new)
                 .flatMap(retryItem -> {
+                    if(retryWhenCondition != null && !retryWhenCondition.get()) {
+                        String errorMessage = String.format(
+                                "Retry condition not met for %s. Last error: %s. Returning an error to the caller",
+                                title, retryItem.cause.getMessage()
+                        );
+                        return Observable.error(new IOException(errorMessage, retryItem.cause));
```

Comment:
minor: why `IOException`?

Reply:
Retries are used mostly in connectors so `IOException` seemed reasonable.